contracts/reverseRegistrar/SignatureReverseResolver.sol

```diff
@@ -0,0 +1,258 @@
+pragma solidity >=0.8.4;
+
+import "../registry/ENS.sol";
+import "./ISignatureReverseResolver.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "../root/Controllable.sol";
+import "../utils/LowLevelCallUtils.sol";
+
+error InvalidSignature();
+error SignatureOutOfDate();
+error Unauthorised();
+
+contract SignatureReverseResolver is Ownable, ISignatureReverseResolver {
+    using ECDSA for bytes32;
+    mapping(bytes32 => uint256) public lastUpdated;
+    mapping(uint64 => mapping(bytes32 => mapping(string => string))) versionable_texts;
+    mapping(uint64 => mapping(bytes32 => string)) versionable_names;
+    mapping(bytes32 => uint64) internal recordVersions;
+
+    bytes32 public immutable parentNode;
+    uint256 public immutable coinType;
```

Comment:
We should just use chain ID here.

Reply:
`coinType` was introduced by @jefflau when he first created `isAuthorisedWithSignature` so needs his input on why `coinType` was used over `chainId`.

Reply:
I think changing to chainId should be safe. I had it as cointype so we could use our derived cointype ids and possibly use this contract for non-EVM L2 chains in the future.